src/services/messageBoard.ts

```diff
@@ -0,0 +1,136 @@
+import { Pool } from "pg";
+import { Request, Response } from "express";
+import config from "config";
+
+const addressesRequestLimit: number = config.get("server.addressRequestLimit");
+export interface Message {
+  [key: string]: {
+    block_no: number;
+    title: string;
+    content: string;
+    valid: number | null;
+    expire: number | null;
+  };
+}
+
+export interface MessageJson {
+  [key: string]: {
+    title: string;
+    content: string[];
+    valid: number | null;
+    expire: number | null;
+  };
+}
+
+interface Dictionary<T> {
+  [keys: string]: T;
+}
+
+export const handleMessageBoard =
+  (p: Pool) =>
+  async (req: Request, res: Response): Promise<void> => {
+    if (!req.body.poolIds) throw new Error("No poolIds in body");
+    const hashes = req.body.poolIds;
+
+    if (!(hashes instanceof Array) || hashes.length > addressesRequestLimit)
+      throw new Error(
+        ` poolIds must have between 0 and ${addressesRequestLimit} items`
+      );
```

Comment:
Confusing that `poolId` field is required, but allowed to be an empty array at the same time. I'd rather require it to be non-empty